examples/backup_vm.py

```diff
@@ -439,11 +456,14 @@ def download_disk(connection, backup_uuid, disk, disk_path, args, incremental=Fa
                 secure=args.secure,
                 buffer_size=args.buffer_size,
                 progress=pb,
+                backing_file=backing_file,
+                backing_format="qcow2",
                 **extra_args)
     finally:
         progress("Finalizing image transfer")
         imagetransfer.finalize_transfer(connection, transfer, disk)
-        progress("Download completed successfully")
+
+    progress("Download completed successfully")
```

Comment:
The scope of the try block is from the point we created a transfer, until we finished
using it. Whatever happens in the scope must end in finalizing the transfer.

This is what we have in this scope:

        progress("Image transfer %r is ready" % transfer.id)
        download_url = transfer.transfer_url

        extra_args = {}
        parameters = inspect.signature(client.download).parameters

        # Use multiple workers to speed up the download.
        if "max_workers" in parameters:
            extra_args["max_workers"] = args.max_workers

        # Use proxy_url if available. Download will use proxy_url if
        # transfer_url is not available.
        if "proxy_url" in parameters:
            extra_args["proxy_url"] = transfer.proxy_url

        with client.ProgressBar() as pb:
            client.download(
                download_url,
                disk_path,
                args.cafile,
                incremental=incremental,
                secure=args.secure,
                buffer_size=args.buffer_size,
                progress=pb,
                **extra_args)

Can you explain which lines should not be there, and how they can raise?

And again, how this this related to the fix added by this commit?

Reply:
You wrote that this PR also handles `some minor improvements to ... error handling`.
Similar (IMHO) to the improvement/fix that you added that you don't want `Download completed successfully` to be printed in `finally` (which can happen even in exception raised flow, so obviously it's not `Download completed successfully`) I thought that you can also remove the following code to be before the `try` block since in my opinion it can't raise an exception and then better be before the `try` block.
```
        progress("Image transfer %r is ready" % transfer.id)
        download_url = transfer.transfer_url

        extra_args = {}

        parameters = inspect.signature(client.download).parameters

        # Use multiple workers to speed up the download.
        if "max_workers" in parameters:
            extra_args["max_workers"] = args.max_workers

        # Use proxy_url if available. Download will use proxy_url if
        # transfer_url is not available.
        if "proxy_url" in parameters:
            extra_args["proxy_url"] = transfer.proxy_url

```
Anyway, that was a suggestion, I already approved.

Reply:
I responded yesterday, strange that I don't see it now...
Anyway, responding again, so it will not look like I ignored your question :)

You wrote that this PR also handles `some minor improvements to ... error handling`.
Similar (IMHO) to the improvement/fix that you added that you don't want `Download completed successfully` to be printed in `finally` (which can happen even in exception raised flow, so obviously it's not `Download completed successfully`) I thought that you can also remove the following code to be before the `try` block since in my opinion it can't raise an exception and then better be before the `try` block.
```
        progress("Image transfer %r is ready" % transfer.id)
        download_url = transfer.transfer_url

        extra_args = {}

        parameters = inspect.signature(client.download).parameters

        # Use multiple workers to speed up the download.
        if "max_workers" in parameters:
            extra_args["max_workers"] = args.max_workers

        # Use proxy_url if available. Download will use proxy_url if
        # transfer_url is not available.
        if "proxy_url" in parameters:
            extra_args["proxy_url"] = transfer.proxy_url
```
Anyway, that was a suggestion, I already approved.